src/platform/cert_capi_openssl.c

```diff
@@ -39,6 +39,63 @@
 #include <wincrypt.h>
 #include <msquic.h>
 
+BOOLEAN
+CxPlatTlsVerifyCertificate(
+    _In_ X509* X509Cert,
+    _In_ const char* SNI
+    )
+{
+    // Convert SNI to wide
+    BOOLEAN Result = FALSE;
+    PCCERT_CONTEXT CertContext = NULL;
+    unsigned char* OpenSSLCertBuffer = NULL;
+    int OpenSSLCertLength = 0;
+
+    OpenSSLCertLength = i2d_X509(X509Cert, &OpenSSLCertBuffer);
+    if (OpenSSLCertLength <= 0) {
+        QuicTraceEvent(
+            LibraryError,
+            "[ lib] ERROR, %s.",
+            "i2d_X509 failed");
+        goto Exit;
+    }
+
+    CertContext =
+        (PCCERT_CONTEXT)CertCreateContext(
```

Comment:
Personal preference:
```suggestion
        (PCCERT_CONTEXT)
        CertCreateContext(
```